From 2bc480938e9eac94bde058e245454699fe994a78 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:39 -0400
Subject: [tools] perf: Unify page_size usage

Message-id: <1408462094-14194-111-git-send-email-jolsa@redhat.com>
Patchwork-id: 88081
O-Subject: [PATCH RHEL7.1 BZ1131394 110/325] perf tools: Unify page_size usage
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 918512b435e15fefe609d236e0ecd62cb8f389c9
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Sep 12 18:39:35 2013 +0200

description
===========
Making page_size global from the util object.

Removing the not needed one.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index 8469e1c..4ef1013 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -70,7 +70,6 @@ struct perf_record {
  struct perf_session *session;
  const char  *progname;
  int   output;
- unsigned int  page_size;
  int   realtime_prio;
  bool   no_buildid;
  bool   no_buildid_cache;
@@ -119,7 +118,7 @@ static int perf_record__mmap_read(struct perf_record *rec,
 {
  unsigned int head = perf_mmap__read_head(md);
  unsigned int old = md->prev;
- unsigned char *data = md->base + rec->page_size;
+ unsigned char *data = md->base + page_size;
  unsigned long size;
  void *buf;
  int rc = 0;
@@ -360,8 +359,6 @@ static int __cmd_record(struct perf_record *rec, int argc, const char **argv)
 
  rec->progname = argv[0];
 
- rec->page_size = sysconf(_SC_PAGE_SIZE);
-
  on_exit(perf_record__sig_exit, rec);
  signal(SIGCHLD, sig_handler);
  signal(SIGINT, sig_handler);
diff --git a/tools/perf/perf.c b/tools/perf/perf.c
index 245020c..6265778 100644
--- a/tools/perf/perf.c
+++ b/tools/perf/perf.c
@@ -456,6 +456,7 @@ int main(int argc, const char **argv)
 {
  const char *cmd;
 
+ /* The page_size is placed in util object. */
  page_size = sysconf(_SC_PAGE_SIZE);
 
  cmd = perf_extract_argv0_path(argv[0]);
diff --git a/tools/perf/util/python.c b/tools/perf/util/python.c
index 2ac4bc9..07b0fd5 100644
--- a/tools/perf/util/python.c
+++ b/tools/perf/util/python.c
@@ -1038,6 +1038,7 @@ PyMODINIT_FUNC initperf(void)
      pyrf_cpu_map__setup_types() < 0)
   return;
 
+ /* The page_size is placed in util object. */
  page_size = sysconf(_SC_PAGE_SIZE);
 
  Py_INCREF(&pyrf_evlist__type);
-- 
1.7.1